crates/sui-core/src/authority_server.rs

```diff
@@ -838,8 +862,19 @@ impl ValidatorService {
                 .collect()
         };
 
-        for (idx, consensus_position) in transaction_indexes.into_iter().zip(consensus_positions) {
-            results[idx] = Some(SubmitTxResult::Submitted { consensus_position });
+        if is_ping_request {
+            // For ping requests, return the special consensus position.
+            assert_eq!(consensus_positions.len(), 1);
+            results.push(Some(SubmitTxResult::Submitted {
+                consensus_position: consensus_positions[0],
+            }));
```

Comment:
Initially I had it like this, but wanted to make this more seamless to the rest of the code and avoid adding exception handling - but I see you are also aligned in this , so let's keep it